opentechinstitute/commotion-router
# Issue73

Addresses commotion-openwrt issue73: https://github.com/opentechinstitute/commotion-openwrt/issues/73

NOTE that this also depends on https://github.com/opentechinstitute/commotiond/pull/61

TO TEST:

1a) Before running quickstart, type 'thisnode' into your browser. You should get the admin page.
Do not run quickstart yet.

1b) Telnet into the node. 
$ logread | grep commotion.hotplug.thisnode

You should see:
"Adding thisnode to /etc/hosts"
"Checking for UCI entry for thisnode alias"
"No entry found. Creating UCI entry for 'thisnode' alias"
"Setting 'thisnode' ip address to 169.254.xx.xx"

1c) Go back to your browser. Run quickstart.

1d) Log into the node again.
$ logread | grep commotion.hotplug.thisnode

You should see:
"Found thisnode in /etc/hosts"
"Checking for UCI entry for thisnode alias"
"thisnode alias exists with ip address 169.254.xx.xx"

Since we're here, let's just do a sanity check:

$ less /etc/hosts

You should see an entry for 'thisnode'

$ less /etc/config/network

You should see a config section for 'alias'
The type is client, ip addr is 169.254.xx.xx, proto is static, netmask is 255.255.0.0

You should ALSO see:

config interface 'client'
option proto 'commotion'
option type 'bridge'

2) ........

3) PROFIT!
